nats-compiler-plugin/src/main/java/io/ballerina/stdlib/nats/plugin/NatsServiceValidator.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    public void validateNonNatsFunction(FunctionDefinitionNode functionDefinitionNode,
+                                        SyntaxNodeAnalysisContext context) {
+        if (isRemoteFunction(context, functionDefinitionNode)) {
+            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_REMOTE_FUNCTION,
+                    DiagnosticSeverity.ERROR, functionDefinitionNode.location()));
+        }
+    }
+
+    private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
+        SemanticModel semanticModel = context.semanticModel();
+        ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) context.node();
+        Optional<Symbol> symbol = semanticModel.symbol(serviceDeclarationNode);
+        if (symbol.isPresent()) {
+            ServiceDeclarationSymbol serviceDeclarationSymbol = (ServiceDeclarationSymbol) symbol.get();
+            Optional<ServiceAttachPoint> attachPoint = serviceDeclarationSymbol.attachPoint();
+            List<AnnotationSymbol> symbolList = serviceDeclarationSymbol.annotations();
+            if (attachPoint.isEmpty()) {
+                if (symbolList.isEmpty()) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.NO_ANNOTATION,
+                            DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
+                } else if (symbolList.size() > 1) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION_NUMBER,
+                            DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
+                } else {
+                    validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(),
+                            context);
+                }
+            } else {
+                if (attachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
+                    if (serviceDeclarationSymbol.annotations().isEmpty()) {
+                        context.reportDiagnostic(PluginUtils.getDiagnostic(
+                                CompilationErrors.INVALID_SERVICE_ATTACH_POINT,
+                                DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
+                    } else {
+                        validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(),
+                                context);
+                    }
+                }
+            }
+        }
+    }
+
+    private void validateAnnotation(AnnotationSymbol annotationSymbol, Location location,
+                                    SyntaxNodeAnalysisContext context) {
+        Optional<ModuleSymbol> moduleSymbolOptional = annotationSymbol.getModule();
+        if (moduleSymbolOptional.isEmpty()) {
+            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION,
+                    DiagnosticSeverity.ERROR, location));
```

Comment:
This particular case should never happen right? If it does, that'd be a bug in the compiler 🤔